src/lib.rs

```diff
@@ -1,3 +1,69 @@
+//! A library that provides `print!`, `println!`, `dbg!` implementations and
+//! logging capabilities for Espressif devices.
+//!
+//! - Supports all Espressif devices.
+//! - Supports different communication methods:
+//!     - UART (Default)
+//!     - JTAG-Serial (Only available in ESP32-C3, ESP32-C6, ESP32-H2, ESP32-S3)
+//!     - [RTT]
+//!     - No-op: Turns printing into a no-op
+//! - Supports [`defmt`] backend
+//!
+//! # Using `esp-println` as a Library:
+//!
+//! ```toml
+//! esp-println = { version = "0.7.0", features = ["esp32c2"] }
+//! ```
+//! or `cargo add esp-println --features esp32c2`
+//!
+//! Its important to specify your target as feature.
+//!
+//! # Features
+//!
+//! - There is one feature for each supported target: `esp32`, `esp32c2`,
+//!   `esp32c3`, `esp32c6`, esp32h2`, `esp32s2`, `esp32s3` and `esp8266`.
+//!    - One of these features must be enabled.
+//!    - Only one of these features can be enabled at a time.
+//! - There is one feature for each supported communication method: `uart`,
+//!   `jtag-serial` and `rtt` `no-op`.
+//!     - Only one of these features can be enabled at a time.
+//! - `critical-section` enables critical sections.
+//! - `colors` enable colored output.
```

Comment:
`colors` only affects `log` I believe. We might want to add this as a note here, but we could also add a warning to `build.rs` to make this obvious. Thoughts?

Reply:
From what I've seen, you are right. So I added the `build.rs` check and updated the documentation